backtracking/subset_sum.cpp

```diff
@@ -0,0 +1,107 @@
+/**
+ * @file
+ * @brief We are given with an array and a sum value. The algorithms find all
+ * the subsets of that array with sum equal to given sum and return such subsets
+ * count. This approach will have exponential time complexity.
+ *
+ * @details Subset problem (https://en.wikipedia.org/wiki/Subset_sum_problem)
+ * @author [Swastika Gupta](https://github.com/swastyy)
+ */
+
+#include <cassert>   /// for assert
+#include <iostream>  /// for io operations
+#include <vector>    /// for std::vector
+
+/**
+ * @namespace backtracking
+ * @brief subset sum algorithm
+ */
+namespace backtracking {
+/**
+ * @namespace Subsets
+ * @brief Functions for counting subsets(both continuous and non-continuous
+ * subarrays) in a given array with a given sum Time Complexity: O(n * 2^n),
+ * where ‘n’ is the number of elements in the given array.
+ */
+namespace Subsets {
+/**
+ * @brief The main function implements count of subsets
+ * @param sum is the required sum of any subset
+ * @param in_arr is the input array
+ * @returns count of the number of subsets with required sum
+ */
+
+std::uint64_t subset_sum(int sum, const std::vector<int> &in_arr) {
```

Comment:
You can use `uint64_t` instead of `std::uint64_t`.
```suggestion
uint64_t subset_sum(int sum, const std::vector<int> &in_arr) {
```
Also, remember to change the parameters to use `uint`-based types instead.